ql/src/java/org/apache/hadoop/hive/ql/metadata/HiveMaterializedViewsRegistry.java

```diff
@@ -421,13 +420,7 @@ private static RelNode createMaterializedViewScan(HiveConf conf, Table viewTable
     }
 
     // 1.3 Build row type from field <type, name>
-    RelDataType rowType;
-    try {
-      rowType = TypeConverter.getType(cluster, rr, null);
-    } catch (CalciteSemanticException e) {
-      // Bail out
-      return null;
-    }
+    RelDataType rowType = TypeConverter.getType(cluster, rr, null);
```

Comment:
Calcite exception does not need to handle anymore? I guess the internal exception will throw in a higher level, and does not need to handle the null. But I want to highlight, this could be behavior change if the caller handle null.

Reply:
btw it is more cleaner. :)

Reply:
`CalciteSemanticException` is a checked exception so we were forced to catch and handle it here. However, after the changes in this PR it can't be thrown anymore from `TypeConverter.getType` methods because we are able to handle all type conversions.

The caller will never get back null as a result so it doesn't need to handle it.